FirebasePerformance/Sources/Instrumentation/Network/FPRNSURLSessionInstrument.m

```diff
@@ -405,8 +405,12 @@ void InstrumentUploadTaskWithRequestFromData(FPRNSURLSessionInstrument *instrume
       ThrowExceptionBecauseInstrumentHasBeenDeallocated(selector, instrumentor.instrumentedClass);
     }
     typedef NSURLSessionUploadTask *(*OriginalImp)(id, SEL, NSURLRequest *, NSData *);
+    // To avoid a runtime warning in Xcode 15, the given `URLRequest`
+    // should have a nil `HTTPBody`. To workaround this, the `HTTPBody` data is removed.
+    NSMutableURLRequest *requestWithoutHTTPBody = [request mutableCopy];
+    requestWithoutHTTPBody.HTTPBody = nil;
```

Comment:
Are we assuming that there would not be an HTTPBody? Since this is an upload request, I'm concerned that there could be an upload content in the HTTP request. Can we make it to be NIL only when the request does not contain a body?

Reply:
Especially for the backward compatibility scenarios.

Reply:
Is the latest change what you mean?  Does it mean anything if both are non-nil?

Note the comparable fix from July https://github.com/google/GoogleUtilities/pull/114

Reply:
Yes, the feedback was on the latest change. But the new fix makes more sense to me. Thanks for changing it.